--- fortune-mod-3.18.0/fortune/fortune.c.orig	2023-03-07 20:29:28.209801870 +0100
+++ fortune-mod-3.18.0/fortune/fortune.c	2023-03-07 20:41:20.522609607 +0100
@@ -240,11 +240,11 @@ static char *program_version(void)
 static void __attribute__((noreturn)) usage(void)
 {
     (void)fprintf(stderr, "%s\n", program_version());
-    (void)fprintf(stderr, "%s", "fortune [-a");
+    (void)fprintf(stderr, "fortune [-ac");
 #ifdef DEBUG
     (void)fprintf(stderr, "%s", "D");
 #endif /* DEBUG */
-    (void)fprintf(stderr, "%s", "f");
+    (void)fprintf(stderr, "ef");
 #ifdef WITH_REGEX
     (void)fprintf(stderr, "%s", "i");
 #endif
--- fortune-mod-3.18.0/fortune/fortune_with_offensive.docbook5.xml.orig	2022-10-28 12:15:20.000000000 +0200
+++ fortune-mod-3.18.0/fortune/fortune_with_offensive.docbook5.xml	2023-03-07 20:54:40.271610329 +0100
@@ -208,6 +208,12 @@ patterns.</para>
   </listitem>
   </varlistentry>
   <varlistentry>
+  <term><emphasis role='strong' remap='B'>-v</emphasis></term>
+  <listitem>
+  <para>Display version information and exit.</para>
+  </listitem>
+  </varlistentry>
+  <varlistentry>
   <term><emphasis role='strong' remap='B'>-w</emphasis></term>
   <listitem>
 <para>Wait before termination for an amount of time calculated from the
--- fortune-mod-3.18.0/fortune/fortune_without_offensive.docbook5.xml.orig	2022-10-28 12:15:19.000000000 +0200
+++ fortune-mod-3.18.0/fortune/fortune_without_offensive.docbook5.xml	2023-03-07 20:55:09.988116007 +0100
@@ -174,6 +174,12 @@ patterns.</para>
   </listitem>
   </varlistentry>
   <varlistentry>
+  <term><emphasis role='strong' remap='B'>-v</emphasis></term>
+  <listitem>
+  <para>Display version information and exit.</para>
+  </listitem>
+  </varlistentry>
+  <varlistentry>
   <term><emphasis role='strong' remap='B'>-w</emphasis></term>
   <listitem>
 <para>Wait before termination for an amount of time calculated from the
--- fortune-mod-3.18.0/fortune/fortune_with_offensive.template.man.orig	2022-10-28 12:15:20.000000000 +0200
+++ fortune-mod-3.18.0/fortune/fortune_with_offensive.template.man	2023-03-07 20:57:03.934165375 +0100
@@ -155,6 +155,13 @@ patterns\&.
 
 .RE
 .PP
+\fB\-v\fR
+.RS 4
+
+Display version information and exit\&.
+
+.RE
+.PP
 \fB\-w\fR
 .RS 4
 
--- fortune-mod-3.18.0/fortune/fortune_without_offensive.template.man.orig	2022-10-28 12:15:20.000000000 +0200
+++ fortune-mod-3.18.0/fortune/fortune_without_offensive.template.man	2023-03-07 20:57:37.043986004 +0100
@@ -132,6 +132,13 @@ patterns\&.
 
 .RE
 .PP
+\fB\-v\fR
+.RS 4
+
+Display version information and exit\&.
+
+.RE
+.PP
 \fB\-w\fR
 .RS 4
 
