diff -burN fortune-mod-1.99.1.orig/Makefile fortune-mod-1.99.1/Makefile
--- fortune-mod-1.99.1.orig/Makefile	2004-03-06 22:59:19.477538704 +0100
+++ fortune-mod-1.99.1/Makefile	2004-03-06 22:59:33.726372552 +0100
@@ -29,7 +29,7 @@
 FORTMANDIR=$(prefix)/usr/share/man/man6
 FORTMANEXT=6
 # Do we want to install the offensive files? (0 no, 1 yes)
-OFFENSIVE=1
+OFFENSIVE=0
 # Do we want to install files with html tags? (0 no, 1 yes)
 # (Note: These files are not yet available)
 WEB=0
